### ISSUE TITLE
libcurl.so.4 not found during runtime on solaris 10

### ISSUE THREAD

**Issue by shoubamzlibap** [SpiderLabs/ModSecurity#845]
I build my custom apache 2.2.X with custom ModSecurity on solaris 10. Until 2.8.0, the build works fine. Now when switching to ModSecurity 2.9.0 (keeping everything else fixded!), I can still build, but when
I try to load the module, I get

Cannot load /local/apache/modules/mod_security2.so into server: ld.so.1: httpd: fatal: libcurl.so.4: open failed: No such file or directory

I checked and libcurl.so.4 is in the location where it should be. Version 2.8.0 was able to find it.

**Comment by zimmerle** [comment]
Hi @shoubamzlibap,

I don't think 2.8.0 is looking for libcurl. That is a dependency of 2.9.0. 

My guess is that it is a filesystem permission problem or your LD_LIBRARY_PATH is not pointing to the right directory. In any case, you can disable the curl dependency during the build time it is not mandatory.

**Comment by shoubamzlibap** [comment]
Hm. configure for 2.8.0 sports a --with-curl option, but apparently libcurl does not get built in:
For 2.8.0, the command
ldd mod_security2.so |grep curl
has no output. I will check the 2.9 module, and will also try configuring with "'--without-curl".

**Comment by shoubamzlibap** [comment]
I configured 2.9.0 with --with-curl=/local/curl, prefixed LD_LIBRARY_PATH with /local/curl/lib, and made sure the libs are readable by the compiling user. Still, the wrong libcurl gets linked (/usr/local/lib/libcurl.so4). Could this be an issue with the configure script?

Anyway, I was able to build mod_security without curl, using --without-curl, so at least for me the issue is fixe. Thank you @zimmerle !
